scripts/publish-docker.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+set -e
+set -x
+
+if [ ! "$#" -ne 4 ] ; then
+    echo "usage: $0 LOG_NAME BASE_TAG HASH VERSION" >&2
+    exit 1
+fi
+
+log_name="$1"
+base_tag="$2"
+hash="$3"
+version="$4"
+
+underscore_name="$(echo -n "$log_name" | tf - _)"
+
+docker push "$base_tag-$hash"
+docker push "$base_tag-$version"
+
+mkdir -p "/tmp/artifacts"
+
+log_file="/tmp/artifacts/publish-$log_name"
+
+echo "${underscore_name}_hashed: $base_tag-$hash" > "$log_file"
```

Comment:
non-blocking: Making them all `>>` _would_ look a bit more consistent and avoid the risk of accidentally doing a bad copy in the future and ending up with two `>` lines. You can add a `: > "$log_file"` to explicitly truncate the file if there's any chance it might already exist. But this is pretty okay as it is.

You could also have `exec > "$log_file"` and then `echo` without redirection after that.

Reply:
I just ran the `echo`s in a subshell to avoid this entirely.